egui/src/widgets/plot/legend.rs

```diff
@@ -0,0 +1,124 @@
+use std::string::String;
+
+use crate::*;
+
+pub(crate) struct LegendEntry {
+    pub text: String,
+    pub color: Color32,
+    pub checked: bool,
+    pub hovered: bool,
+}
+
+impl LegendEntry {
+    pub fn new(text: String, color: Color32, checked: bool) -> Self {
+        Self {
+            text,
+            color,
+            checked,
+            hovered: false,
+        }
+    }
+}
+
+struct LegendEntryWidget<'a> {
+    entry: &'a mut LegendEntry,
+}
+
+impl<'a> LegendEntryWidget<'a> {
+    fn new(entry: &'a mut LegendEntry) -> Self {
+        Self { entry }
+    }
+}
+
+impl<'a> Widget for LegendEntryWidget<'a> {
+    fn ui(self, ui: &mut Ui) -> Response {
+        let LegendEntry {
+            checked,
+            text,
+            color,
+            ..
+        } = self.entry;
+        let icon_width = ui.spacing().icon_width;
+        let icon_spacing = ui.spacing().icon_spacing;
+        let padding = vec2(2.0, 2.0);
+        let total_extra = padding + vec2(icon_width + icon_spacing, 0.0) + padding;
+
+        let text_style = TextStyle::Button;
+        let galley = ui.fonts().layout_no_wrap(text_style, text.clone());
+
+        let mut desired_size = total_extra + galley.size;
+        desired_size = desired_size.at_least(ui.spacing().interact_size);
+        desired_size.y = desired_size.y.at_least(icon_width);
+
+        ui.with_layout(Layout::top_down(Align::LEFT), |ui| {
+            let (rect, response) = ui.allocate_exact_size(desired_size, Sense::click());
+            let rect = rect.shrink2(padding);
+
+            response.widget_info(|| {
+                WidgetInfo::selected(WidgetType::RadioButton, *checked, &galley.text)
```

Comment:
nit: This is (functionally) a `Checkbox`, since checking and unchecking it is independent from whether or not its siblings are checked

```suggestion
                WidgetInfo::selected(WidgetType::Checkbox, *checked, &galley.text)
```

Reply:
Ah, makes sense. Changed in 6aaeae7.